libraries/botbuilder-ai/src/qnaMaker.ts

```diff
@@ -206,7 +206,7 @@ export class QnAMaker implements QnAMakerTelemetryClient {
         }
 
         // Log telemetry
-        this.onQnaResults(queryResult, context, telemetryProperties, telemetryMetrics);
+        await this.onQnaResults(queryResult, context, telemetryProperties, telemetryMetrics);
```

Comment:
We'd need something at the functional test level instead of just unit test, because apparently no QnAMessage events were actually filled/logged into Azure Application Insights, because we were missing `await` here

Tested manually to verify this change works

Reply:
Does this need to complete prior to the `this.generateAnswerUtils.emitTraceInfo` promise? If not, I'd suggest doing:
```typescript
await Promise.all([first, second]);
```

Reply:
Also, using Sinon and sandboxing can be a useful way to ensure the proper calls occur. It's not a panacea, but if you mocked a method call that occurs inside the promise that this method returns and then invoke `sandbox.verify()` you can ensure the method was called.